src/System.Text.Json/tests/Utf8JsonReaderTests.TryGet.cs

```diff
@@ -263,9 +290,202 @@ public static void TestingNumbersInvalidConversionToInt64(string jsonString, dou
+            Assert.Equal(json.BytesConsumed, json.CurrentState.BytesConsumed);
+        }
+
+        [Theory]
+        [InlineData("-4.402823E+38", float.NegativeInfinity)] // float.MinValue - 1
+        [InlineData("4.402823E+38", float.PositiveInfinity)]  // float.MaxValue + 1
+        public static void TestingNumbersInvalidConversionToSingle(string jsonString, float expected)
+        {
+            byte[] dataUtf8 = Encoding.UTF8.GetBytes(jsonString);
+
+            var json = new Utf8JsonReader(dataUtf8, true, default);
+            while (json.Read())
+            {
+                if (json.TokenType == JsonTokenType.Number)
+                {
+                    #region TryGet calls
+                    Assert.True(json.TryGetSingle(out float actual));
+                    Assert.Equal(expected, actual);
+                    #endregion
+
+                    #region Get calls
+                    Assert.Equal(expected, json.GetSingle());
+                    #endregion
+                }
+            }
+
+            Assert.Equal(dataUtf8.Length, json.BytesConsumed);
+            Assert.Equal(json.BytesConsumed, json.CurrentState.BytesConsumed);
+        }
+
+        [Theory]
+        [InlineData("-2.79769313486232E+308", double.NegativeInfinity)] // double.MinValue - 1
+        [InlineData("2.79769313486232E+308", double.PositiveInfinity)]  // double.MaxValue + 1
+        public static void TestingNumbersInvalidConversionToDouble(string jsonString, double expected)
+        {
+            byte[] dataUtf8 = Encoding.UTF8.GetBytes(jsonString);
+
+            var json = new Utf8JsonReader(dataUtf8, true, default);
+            while (json.Read())
+            {
+                if (json.TokenType == JsonTokenType.Number)
+                {
+                    #region TryGet calls
+                    Assert.True(json.TryGetDouble(out double actual));
+                    Assert.Equal(expected, actual);
+                    #endregion
+
+                    #region Get calls
+                    Assert.Equal(expected, json.GetDouble());
+                    #endregion
+                }
+            }
+
+            Assert.Equal(dataUtf8.Length, json.BytesConsumed);
+            Assert.Equal(json.BytesConsumed, json.CurrentState.BytesConsumed);
+        }
+
+        [Theory]
+        [InlineData("-79228162514264337593543950336", -79228162514264337593543950336d)] // decimal.MinValue - 1
+        [InlineData("79228162514264337593543950336", 79228162514264337593543950336d)]  // decimal.MaxValue + 1
+        public static void TestingNumbersInvalidConversionToDecimal(string jsonString, double expected)
+        {
+            byte[] dataUtf8 = Encoding.UTF8.GetBytes(jsonString);
+
+            var json = new Utf8JsonReader(dataUtf8, true, default);
+            while (json.Read())
+            {
+                if (json.TokenType == JsonTokenType.Number)
+                {
+                    #region TryGet calls
```

Comment:
nit: I don't think creating such small regions is worth it and makes the test noisy. I would only keep the region tags for large areas of code (across multiple APIs/etc.) especially those that won't fit onto the screen. These regions around 3-10 lines of code can be removed.